lib/srv/usermgmt.go

```diff
@@ -404,16 +442,23 @@ func (u *HostUserManagement) createUser(name string, ui services.HostUsersInfo)
 		}
 
 		if userOpts.Home != "" {
-			if err := u.backend.CreateHomeDirectory(userOpts.Home, user.Uid, user.Gid); err != nil {
-				return trace.Wrap(err)
+			if _, err := os.Stat(userOpts.Home); os.IsNotExist(err) {
+				log.InfoContext(context.Background(), "Creating home directory", "home", userOpts.Home, "gid", userOpts.GID)
+				if err := u.backend.CreateHomeDirectory(userOpts.Home, user.Uid, user.Gid); err != nil {
```

Comment:
That's essentially what `CreateHomeDirectory` does, but it swallows the error context instead of returning it to the caller. I suppose we could surface the error when the directory already exists and handle it here instead 🤔

Reply:
I dropped the `os.Stat` in favor of making `CreateHomeDirectory` return the `os.ErrExists` so that callers can handle the cases as they see fit :+1: